PurchasesCoreSwift/Public/Offering.swift

```diff
@@ -0,0 +1,127 @@
+     */
+    @objc public let serverDescription: String
+
+    /**
+     Array of `RCPackage` objects available for purchase.
+     */
+    @objc public let availablePackages: [Package]
+
+    /**
+     Lifetime package type configured in the RevenueCat dashboard, if available.
+     */
+    @objc private(set) public var lifetime: Package?
+
+    /**
+     Annual package type configured in the RevenueCat dashboard, if available.
+     */
+    @objc private(set) public var annual: Package?
+
+    /**
+     Six month package type configured in the RevenueCat dashboard, if available.
+     */
+    @objc private(set) public var sixMonth: Package?
+
+    /**
+     Three month package type configured in the RevenueCat dashboard, if available.
+     */
+    @objc private(set) public var threeMonth: Package?
+
+    /**
+     Two month package type configured in the RevenueCat dashboard, if available.
+     */
+    @objc private(set) public var twoMonth: Package?
+
+    /**
+     Monthly package type configured in the RevenueCat dashboard, if available.
+     */
+    @objc private(set) public var monthly: Package?
+
+    /**
+     Weekly package type configured in the RevenueCat dashboard, if available.
+     */
+    @objc private(set) public var weekly: Package?
+
+    // TODO(post-migration): Change back to internal instead of public
+    @objc public init(identifier: String, serverDescription: String, availablePackages: [Package]) {
+        self.identifier = identifier
+        self.serverDescription = serverDescription
+        self.availablePackages = availablePackages
+
+        // TODO: add validation to ensure we don't get multiple packages with the same type
+        for package in availablePackages {
+            switch package.packageType {
+            case .lifetime:
+                self.lifetime = package
+            case .annual:
+                self.annual = package
+            case .sixMonth:
+                self.sixMonth = package
+            case .threeMonth:
+                self.threeMonth = package
+            case .twoMonth:
+                self.twoMonth = package
+            case .monthly:
+                self.monthly = package
+            case .weekly:
+                self.weekly = package
+            case .unknown, .custom:
+                break
+            }
+        }
+    }
+
+    public override var description: String {
+        return """
+        <Offering {\n\tidentifier=\(identifier)\n\tserverDescription=\(serverDescription)\n"
+        \tavailablePackages=\(valueOrEmpty(availablePackages))\n\tlifetime=\(valueOrEmpty(lifetime))\n
+        \tannual=\(valueOrEmpty(annual))\n\tsixMonth=\(valueOrEmpty(sixMonth))\n
+        \tthreeMonth=\(valueOrEmpty(threeMonth))\n\ttwoMonth=\(valueOrEmpty(twoMonth))\n
+        \tmonthly=\(valueOrEmpty(monthly))\n\tweekly=\(valueOrEmpty(weekly))\n}>
+        """
+    }
+
+    private func valueOrEmpty<T: CustomStringConvertible>(_ value: T?) -> String {
```

Comment:
we should move the private stuff to the bottom or an extension

Reply:
💯

Reply:
I moved it to the bottom for now. We should definitely consider standardizing some coding guidelines around extension usage. I don't generally use them for private things, just conformance to protocols. I'm open to doing things differently though.

Reply:
#643 I had started this just before reading this comment! happy to update any of the stuff there, I don't have strong preferences but I do think we should have a standard

Reply:
Fantastic!